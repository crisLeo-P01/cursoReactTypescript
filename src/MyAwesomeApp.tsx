import type { CSSProperties } from "react";

const firstName = "Cristian";
const lastName = "Paiva";

const favoriteGames = ['ETS2', 'Grand Thead Auto V', 'Cyberpunk']

const isActive = false

const address = {
    zipCode: 'ABC-123',
    country: 'Argentina'
}

const estado: CSSProperties = {
    backgroundColor: 'red',
    color: 'white',
    fontSize: 18,
    padding: '10px 20px',
    borderRadius: 20
}

export function MyAwesomeApp() {

    return (
        <div>
            <h1 data-testid="first-name-title">{firstName}</h1>
            <h3>{lastName}</h3>

            <p className="p-favorita" style={{
                fontSize: 16,
                color: '#161616',
                backgroundColor: '#f1f1f1',
                padding: "5px 20px",
                borderRadius: 20
            }}>{favoriteGames.join(', ')}</p>

            <h3 style={estado}>{isActive ? 'Activo' : 'No activo'}</h3>

            <p>{JSON.stringify(address)}</p>
        </div>
    )
}