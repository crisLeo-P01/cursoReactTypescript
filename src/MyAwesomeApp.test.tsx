import { describe, expect, test } from "vitest";
import { render, screen } from '@testing-library/react'

import {MyAwesomeApp} from './MyAwesomeApp'

describe('MyAwesomeApp', () => { // El describe es para agrupar tests
    test('Renderizar firstName y lastName', () => { // El test es para definir un caso de prueba
        const { container } = render(<MyAwesomeApp />)

        screen.debug() // Muestra el HTML renderizado en la consola

        const h1 = container.querySelector('h1')
        const h3 = container.querySelector('h3')

        expect(h1?.innerHTML).toContain('Cristian') // El ? es para evitar errores si h1 es null
        expect(h3?.innerHTML).toContain('Paiva') // El toContain es para verificar que el string contiene el valor esperado

    });

    test('Renderizar firstName y lastName', () => {
        render(<MyAwesomeApp />) // El render es para renderizar el componente

        screen.debug()

        const h1 = screen.getByTestId('first-name-title')

        expect(h1?.innerHTML).toContain('Cristian') // El expect es para hacer una aserciÃ³n

    });

    test('Crear un snapshot de MyAwesomeApp', () => {
        const { container } = render(<MyAwesomeApp />)
        expect(container).toMatchSnapshot()
    })
})