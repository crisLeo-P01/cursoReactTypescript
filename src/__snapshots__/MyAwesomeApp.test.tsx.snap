// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > Crear un snapshot de MyAwesomeApp 1`] = `
<div>
  <div>
    <h1
      data-testid="first-name-title"
    >
      Cristian
    </h1>
    <h3>
      Paiva
    </h3>
    <p
      class="p-favorita"
      style="font-size: 16px; color: rgb(22, 22, 22); background-color: rgb(241, 241, 241); padding: 5px 20px; border-radius: 20px;"
    >
      ETS2, Grand Thead Auto V, Cyberpunk
    </p>
    <h3
      style="background-color: red; color: white; font-size: 18px; padding: 10px 20px; border-radius: 20px;"
    >
      No activo
    </h3>
    <p>
      {"zipCode":"ABC-123","country":"Argentina"}
    </p>
  </div>
</div>
`;
